cudapoa/src/cudapoa_batch.cpp

```diff
@@ -312,6 +312,85 @@ StatusType CudapoaBatch::get_msa(std::vector<std::vector<std::string>>& msa, std
+                                     cudaMemcpyDeviceToHost,
+                                     stream_));
+
+    CGA_CU_CHECK_ERR(cudaMemcpyAsync(graph_details_h_->incoming_edge_weights,
+                                     graph_details_d_->incoming_edge_weights,
+                                     sizeof(uint16_t) * max_nodes_per_window_ * CUDAPOA_MAX_NODE_EDGES * max_poas_,
+                                     cudaMemcpyDeviceToHost,
+                                     stream_));
+
+    CGA_CU_CHECK_ERR(cudaMemcpyAsync(graph_details_h_->incoming_edge_count,
+                                     graph_details_d_->incoming_edge_count,
+                                     sizeof(uint16_t) * max_nodes_per_window_ * max_poas_,
+                                     cudaMemcpyDeviceToHost,
+                                     stream_));
+
+    CGA_CU_CHECK_ERR(cudaMemcpyAsync(input_details_h_->sequence_lengths,
+                                     input_details_d_->sequence_lengths,
+                                     global_sequence_idx_ * sizeof(uint16_t),
+                                     cudaMemcpyDeviceToHost,
+                                     stream_));
+
+    CGA_CU_CHECK_ERR(cudaMemcpyAsync(output_details_h_->consensus,
+                                     output_details_d_->consensus,
+                                     CUDAPOA_MAX_CONSENSUS_SIZE * max_poas_ * sizeof(uint8_t),
+                                     cudaMemcpyDeviceToHost,
+                                     stream_));
+
+    // Reservet host space for graphs
+    graphs.resize(poa_count_);
+
+    CGA_CU_CHECK_ERR(cudaStreamSynchronize(stream_));
+
+    for (int32_t poa = 0; poa < poa_count_; poa++)
+    {
+        char* c = reinterpret_cast<char*>(&(output_details_h_->consensus[poa * CUDAPOA_MAX_CONSENSUS_SIZE]));
+        // We use the first two entries in the consensus buffer to log error during kernel execution
+        // c[0] == 0 means an error occured and when that happens the error type is saved in c[1]
+        if (static_cast<uint8_t>(c[0]) == CUDAPOA_KERNEL_ERROR_ENCOUNTERED)
+        {
+            decode_cudapoa_kernel_error(static_cast<claragenomics::cudapoa::StatusType>(c[1]), output_status);
+        }
+        else
+        {
+            output_status.emplace_back(claragenomics::cudapoa::StatusType::success);
+            DirectedGraph& graph = graphs[poa];
+            int32_t seq_0_offset = input_details_h_->window_details[poa].seq_len_buffer_offset;
+            int32_t num_nodes    = input_details_h_->sequence_lengths[seq_0_offset];
+            uint8_t* nodes       = &graph_details_h_->nodes[max_nodes_per_window_ * poa];
+            for (int32_t n = 0; n < num_nodes; n++)
+            {
+                // For each node, find it's incoming edges and add the edge to the graph,
+                // along with its label.
+                Graph::node_id_t sink = n;
```

Comment:
Remove the `Graph` class here, and change the types to `DirectedGraph::node_id_t`, ...